packages/account/src/hdwallet/hdwallet.ts

```diff
@@ -221,7 +221,8 @@ class HDWallet {
   }
 
   static fromExtendedKey(extendedKey: string) {
-    const decoded = toBeHex(decodeBase58(extendedKey));
+    let decoded = toHex(decodeBase58(extendedKey));
+    decoded = `${decoded.substring(0, 2)}0${decoded.substring(2)}`;
```

Comment:
This was a strange addition on my part needed to make the tests pass, essentially toHex omits a padded 0 at the third index, thus 0x0488b21e03587302... becomes 0x488b21e03587302.... but I would think that if toHex had a bug it would've been discovered much earlier... any thoughts?

Reply:
I'm not sure I agree with that, but regardless the [`toHex` function](https://github.com/FuelLabs/fuels-ts/blob/refactor/internalize-ethers-deprecate-getnetwork/packages/math/src/functional.ts#L15) was implemented some time ago is used pretty widely across the repo so adjusting it would be a substantial undertaking beyond the scope of this PR.

Reply:
cc @nedsalk — is this relevant to the problems you had with `toHex` vs. `hexlify`?

Reply:
I had an issue where I was using `toHex(randomBytes(32))` instead of `hexlify(randomBytes(32))` for generating asset ids in a custom chain config, which the `fuel-core` node sometimes accepted and sometimes couldn't interpret as a valid asset id. After switching to `hexlify`, the problem went away. From the names of the functions, one would expect the same behavior; I didn't investigate what their differences are.

@maschad Did you try `hexlify(decodeBase58(extendedKey))` here?

Reply:
@nedsalk thanks for the suggestion, I actually ended up converting it to bytes then hexilify-ing it.

Reply:
@maschad Is `toHex` still omiting the leading zero?

Reply:
That's my understanding, but I am using `hexlify` here